### PR TITLE
Bookmark explore listing

### PR DESCRIPTION
**Pull request by abhishektanwar**
FEATURES ADDED:
- Explore page
- Bookmarks page

### CODE REVIEW

**Review by pnchinmay — APPROVED** [review]
Code is modular
UI looks clean
Good to go :100:

File: src/components/Feed/Feed.jsx [hunk 1]
```diff
@@ -10,25 +10,35 @@ import "./feed.css";
 const Feed = ({ posts, status, error }) => {
   const { user } = useSelector((state) => state.user);
   const { userId } = useParams();
+  const { pathname } = useLocation();
+  console.log("location", pathname);
```
Comment (zoyatanweer) [inline src/components/Feed/Feed.jsx]: remove console.log

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -40,8 +40,8 @@ const Header = () => {
           onClick={() => {authToken ? dispatch(logout()) : navigate('/login')}}
         />
       </div>
-      <div className="nav-section">
-        {/* <BadgeIconButton
+      {/* <div className="nav-section">
```
Comment (zoyatanweer) [inline src/components/Header/Header.jsx]: remove comment if not required

File: src/components/LeftSidebar/LeftSidebar.jsx [hunk 3]
```diff
@@ -6,56 +6,60 @@ import {
...
       onClick={onNavBtnClick}
     />
   );
 };
 
-const LeftSidebar = ({user}) => {
+const LeftSidebar = ({ user }) => {
   const navigate = useNavigate();
+  const { pathname } = useLocation();
+  console.log("pathname", pathname);
```
Comment (zoyatanweer) [inline src/components/LeftSidebar/LeftSidebar.jsx]: remove console.log

File: src/components/LeftSidebar/left-sidebar.css [hunk 4]
```diff
@@ -18,9 +18,18 @@
...
+
 .sidebar-list-item:hover{
   background-color: var(--border-color);
   transform: scale(1.2);
+  max-width: 80%;
+  left: 0;
+  right: 0;
+  margin-left: auto;
+  margin-right: auto;
   /* border-radius:/ 1rem; */
```
Comment (zoyatanweer) [inline src/components/LeftSidebar/left-sidebar.css]: remove comment if not required

File: src/components/Post/Post.jsx [hunk 5]
```diff
@@ -30,11 +30,11 @@ const Post = ({ post }) => {
...
   const [isLiked, setIsLiked] = useState(likes.includes(currentUser?._id));
   const [isBookmarked, setIsBookmarked] = useState(
-    // currentUser?.bookmarks.includes(post._id)
-    true
+    currentUser?.bookmarks.find((bookmark) => bookmark._id === post._id)
   );
   const [comment, setComment] = useState("");
   const [showCommentBox, setShowCommentBox] = useState(false);
+  const navigate = useNavigate();
   // const isLiked = likes.includes(currentUser._id)
```
Comment (zoyatanweer) [inline src/components/Post/Post.jsx]: remove comment if not req

File: src/components/Post/Post.jsx [hunk 6]
```diff
@@ -43,21 +43,26 @@ const Post = ({ post }) => {
   };
 
   const handleBookmarkPost = () => {
-    dispatch(bookmarkPost({ postId: post._id }));
+    // dispatch(bookmarkPost({ postId: post._id }));
```
Comment (zoyatanweer) [inline src/components/Post/Post.jsx]: remove comment

File: src/components/Post/Post.jsx [hunk 7]
```diff
@@ -43,21 +43,26 @@ const Post = ({ post }) => {
...
+    dispatch(bookmarkPost({ postId: post._id, data: post }));
+
     setIsBookmarked((prev) => !prev);
     // setPostLikes((prev) => (isLiked ? prev - 1 : prev + 1));
```
Comment (zoyatanweer) [inline src/components/Post/Post.jsx]: remove comment if not req

File: src/index.css [hunk 8]
```diff
@@ -32,11 +32,17 @@
...
+    margin-right: auto;
 }
 
 .app{
   background-color:var(--grey-background-color);
-  height: 100%;
+  min-height: 100vh;
+  height: 100vh;
+  overflow-x: hidden;
+  /* overflow-y:hidden; */
```
Comment (zoyatanweer) [inline src/index.css]: remove comment

File: src/pages/LandingPage/LandingPage.jsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+          <img
+            loading="lazy"
+            className="responsive-img"
+            src={utils.getImg("landing-page-banner-image.svg")}
+            style={{ maxWidth: "340px" }}
+          />
+        </div>
+      </div>
+      <div className="flex-row margin-top-60 flex-justify-content-space-between">
+        {/* {contentData.map((data) => {
```
Comment (zoyatanweer) [inline src/pages/LandingPage/LandingPage.jsx]: remove if not req

File: src/pages/bookmarks/Bookmarks.jsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    finalFilteredPosts = trendingPostsFilter(user?.bookmarks);
+  } else if (filterType === null) {
+    finalFilteredPosts = user?.bookmarks;
+  }
+  return (
+    <div>
+      <div className="flex-row home-container app-container">
+        <LeftSidebar user={user} />
+        <Feed posts={finalFilteredPosts} status={status} error={error} />
+        {/* <RightSidebar component={<HomeRightbar />} /> */}
```
Comment (zoyatanweer) [inline src/pages/bookmarks/Bookmarks.jsx:29]: remove comment

File: src/pages/explore/Explore.jsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  }
+  useEffect(()=>{
+    dispatch(getExplorePosts())
+  },[])
+  return (
+    <div>
+      <div className="flex-row home-container app-container">
+        <LeftSidebar user={user} />
+        <Feed posts={finalFilteredPosts} status={status} error={error} />
+        {/* <RightSidebar component={<HomeRightbar />} /> */}
```
Comment (zoyatanweer) [inline src/pages/explore/Explore.jsx:33]: remove comment

File: src/pages/home/HomeRightbar.jsx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+const HomeRightbar = () => {
+  const dispatch = useDispatch();
+  const { user, allUsers, fetchingAllUsersStatus } = useSelector(
+    (state) => state.user
+  );
+  const followSuggestionsArr = allUsers?.filter(
+    (ar) => !user?.following?.find((rm) => rm._id === ar._id)
+  );
+  const navigate = useNavigate();
+  // useEffect(() => {
```
Comment (zoyatanweer) [inline src/pages/home/HomeRightbar.jsx:18]: remove if not req

File: src/pages/home/HomeRightbar.jsx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+        )}
+      </div>
+    </div>
+  );
+};
+
+export default HomeRightbar;
+
+{
+  /* <img
```
Comment (zoyatanweer) [inline src/pages/home/HomeRightbar.jsx]: remove if not req

File: src/reducers/postSlice.js [hunk 14]
```diff
@@ -42,6 +43,23 @@ export const loadPosts = createAsyncThunk(
...
 
+export const getExplorePosts = createAsyncThunk(
+  "counter/getExplorePosts",
+  async (_, { rejectWithValue }) => {
+    try {
+      const result = await axios.request({
+        method: "get",
+        url: `https://trekgram-backend.herokuapp.com/api/post/posts/getAllPosts`,
+      });
+      console.log("result", result);
```
Comment (zoyatanweer) [inline src/reducers/postSlice.js]: remove console.log

File: src/reducers/postSlice.js [hunk 15]
```diff
@@ -42,6 +43,23 @@ export const loadPosts = createAsyncThunk(
...
+      return result.data;
+    } catch (err) {
+      console.log("counter/loadPosts", err);
```
Comment (zoyatanweer) [inline src/reducers/postSlice.js]: remove console.log

File: src/reducers/postSlice.js [hunk 16]
```diff
@@ -350,6 +368,22 @@ export const postSlice = createSlice({
...
+                  userId: action.payload.userId,
+                  comment: action.payload.comment,
+                  profilePicture: action.payload.profilePicture,
+                  username: action.payload.username,
+                },
+              ],
+            }
+          : post
+      );
       // state.posts = action.payload;
```
Comment (zoyatanweer) [inline src/reducers/postSlice.js:360]: remove comment

File: src/reducers/postSlice.js [hunk 17]
```diff
@@ -395,6 +434,19 @@ export const postSlice = createSlice({
       state.status = "idle";
       state.errorMessage = action.payload;
     },
+    [getExplorePosts.pending]: (state) => {
+      state.status = "loading";
+    },
+    [getExplorePosts.fulfilled]: (state, action) => {
+      console.log("action posts", action);
```
Comment (zoyatanweer) [inline src/reducers/postSlice.js]: remove console.log

File: src/reducers/userSlice.js [hunk 18]
```diff
@@ -121,6 +123,23 @@ export const unFollowUser = createAsyncThunk("user/unFollowUser",async ({targetU
   }
 });
 
+export const getAllUsers = createAsyncThunk("user/getAllUsers",async(_,{rejectWithValue})=>{
+  try {
+    // console.log("unfollow data",targetUserId,userId);
```
Comment (zoyatanweer) [inline src/reducers/userSlice.js]: remove console

File: src/reducers/userSlice.js [hunk 19]
```diff
@@ -231,6 +277,34 @@ export const userSlice = createSlice({
...
+    },
+    [bookmarkPost.fulfilled]:(state,action)=>{
+      state.fetchingAllUsersStatus = "fulfilled"
+      if(action.payload.response === "Post bookmarkedd"){
+        state.user = {...state.user,bookmarks:[...state.user.bookmarks,action.payload.data]}
+      }
+      else if(action.payload.data === "Post un bookmarked"){
+        state.user = {...state.user,bookmarks:state.user.bookmarks.filter((post)=>post._id !== action.payload.data._id)}
+      }
+      // state.user = {...state.user,bookmarks}
```
Comment (zoyatanweer) [inline src/reducers/userSlice.js:354]: remove if not req

**Review by zoyatanweer — APPROVED** [review]
- code quality is good and easy to understand
- suggestion: you could use comments to show where component is starting
- good to merge with suggested changes

File: src/components/Feed/Feed.jsx (hunk 1 above)
Comment (abhishektanwar) [reply src/components/Feed/Feed.jsx]: removed

File: src/components/Header/Header.jsx (hunk 2 above)
Comment (abhishektanwar) [reply src/components/Header/Header.jsx]: removed

File: src/components/LeftSidebar/LeftSidebar.jsx (hunk 3 above)
Comment (abhishektanwar) [reply src/components/LeftSidebar/LeftSidebar.jsx]: removed.

File: src/components/LeftSidebar/left-sidebar.css (hunk 4 above)
Comment (abhishektanwar) [reply src/components/LeftSidebar/left-sidebar.css]: removed

File: src/components/Post/Post.jsx (hunk 7 above)
Comment (abhishektanwar) [reply src/components/Post/Post.jsx]: removed

File: src/index.css (hunk 8 above)
Comment (abhishektanwar) [reply src/index.css]: removed

File: src/pages/LandingPage/LandingPage.jsx (hunk 9 above)
Comment (abhishektanwar) [reply src/pages/LandingPage/LandingPage.jsx]: removed

File: src/pages/home/HomeRightbar.jsx (hunk 12 above)
Comment (abhishektanwar) [reply src/pages/home/HomeRightbar.jsx:18]: this feature will be coming in future. That is why I have kept it.

File: src/pages/home/HomeRightbar.jsx (hunk 13 above)
Comment (abhishektanwar) [reply src/pages/home/HomeRightbar.jsx]: removed

File: src/reducers/postSlice.js (hunk 14 above)
Comment (abhishektanwar) [reply src/reducers/postSlice.js]: removed

File: src/reducers/postSlice.js (hunk 15 above)
Comment (abhishektanwar) [reply src/reducers/postSlice.js]: removed

File: src/reducers/postSlice.js (hunk 16 above)
Comment (abhishektanwar) [reply src/reducers/postSlice.js:360]: removed

File: src/reducers/userSlice.js (hunk 18 above)
Comment (abhishektanwar) [reply src/reducers/userSlice.js]: removed

File: src/reducers/postSlice.js (hunk 17 above)
Comment (abhishektanwar) [reply src/reducers/postSlice.js]: removed

**Pull request merged**